File: CarPoolSolution/Models/BookedRide.cs

using CarPool.Enums;

namespace CarPool.Models
{
    public class BookedRide
    {        
        public int Id { get; set; }
        public Stop From { get; set; }
        public Stop To { get; set; }
        public string Date { get; set; }
        public string Time { get; set; }
        public int SeatsBooked { get; set; }
        public int OfferId { get; set; }
        public int UserId { get; set; }
        public int Price { get; set; }
    }
}

File: CarPoolSolution/CarPoolAngular/src/app/models/BookedRide.ts
import { Stops } from '../constants/Stops';

export class BookedRide {
  public From!: Stops | string;
  public To!: Stops | string;
  public Date!: string;
  public Time!: string;
  public SeatsBooked!: number;
  public OfferId!: number;
  public userId!: number;
  public Price!: number;

  constructor(
    from: Stops | string,
    to: Stops | string,
    date: string,
    time: string,
    seatsbooked: number,
    price: number,
    userId: number
  ) {
    this.From = from;
    this.To = to;
    this.Date = date;
    this.Time = time;
    this.SeatsBooked = seatsbooked;
    this.Price = price;
    this.userId = userId;
  }
}
